### PR TITLE
Fp16 flops

### PR DESCRIPTION
**Pull request by shwetasalaria**

### CODE REVIEW

File: benchmarker/__main__.py [hunk 1]
```diff
@@ -62,7 +62,10 @@ def main():
         else:
             result["device"] = result["platform"]["cpu"]["brand"]
         if args.flops:
-            result["problem"]["gflop_measured"] = get_counters(command)
+            if 'Intel(R)' in result["device"].split(" "):
+                result["problem"]["gflop_measured"] = get_counters(command)
+            else:
+                result["problem"]["gflop_measured"] = "N/A"
```
Comment (vatai) [inline benchmarker/__main__.py:68]: Do we use `"N/A"` anywhere else? Don't we just omit those fields which we don't have? Won't this cause problems later, in the processing of arguments?!

Do you have to split the `result["device"]`? Can't you do it directly?

I would have just added here `if args.flops and 'Intel' in result["device"]:` and left it as is (+ get counters again doesn't return counters, right?)

**Review by vatai — CHANGES_REQUESTED** [review]
Two critical points:
- the breaking change of checking `"gflops_measured"`
- the overcomplicated checking if the cpu is Intel

File: benchmarker/__main__.py (hunk 1 above)
Comment (shwetasalaria) [reply benchmarker/__main__.py:68]: No, we don't use N/A anywhere else yet. We handle the processing of gflops_measured by checking if it is not a string

File: benchmarker/nvprof.py [hunk 2]
```diff
@@ -1,13 +1,19 @@
 from benchmarker.util import abstractprocess
 
 
-def get_nvprof_counters(command):
-    sep = "flop_count_sp   Floating Point Operations(Single Precision)"
+def get_nvprof_counters(command, precision):
+    if precision == "FP16":
```
[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/nvprof.py:5]: But nvprof has counters for FP16, FP32 and FP64 only

File: benchmarker/__main__.py [hunk 3]
```diff
@@ -51,7 +51,7 @@ def main():
     if result["nb_gpus"] > 0:
         result["device"] = result["platform"]["gpus"][0]["brand"]
         if args.flops:
-            result["problem"]["gflop_measured"] = get_nvprof_counters(command)
+            result["problem"]["gflop_measured"] = get_nvprof_counters(command, result["problem"]["precision"])
```
[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/__main__.py:54]: How about "get_gpu_flops()" and "get_cpu_flops()"?
If it is ok, I can make these changes in this PR.

File: benchmarker/nvprof.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/nvprof.py:5]: Need a check to get gflops when precision is either FP16 or FP32

[the remark this answers is not in the document]
Comment (vatai) [reply benchmarker/nvprof.py:5]: Yeah, it is probably ok.

File: benchmarker/__main__.py [hunk 4]
```diff
@@ -77,7 +80,7 @@ def main():
     cute_device = get_cute_device_str(result["device"]).replace(" ", "_")
     result["path_out"] = os.path.join(result["path_out"], result["problem"]["name"])
     result["path_out"] = os.path.join(result["path_out"], cute_device)
-    if "gflop_measured" in result["problem"]:
+    if not isinstance(result["problem"]["gflop_measured"], str):
```
[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/__main__.py]: Do you mean line 80? I didn't add that piece of code, I think Alex did.

[the remark this answers is not in the document]
Comment (vatai) [reply benchmarker/__main__.py]: So this is reverted, right? If yes, please mark the conversation resolved.

[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/__main__.py]: Yes, this is reverted.

File: benchmarker/__main__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shwetasalaria) [reply benchmarker/__main__.py]: Renamed it to get_gflop()

**Pull request merged**